ert_gui/simulation/view/progress.py

```diff
@@ -0,0 +1,99 @@
+from qtpy.QtCore import QSize, QModelIndex, Slot, Qt
+from qtpy.QtWidgets import (
+    QTreeView,
+    QStyledItemDelegate,
+    QStyleOptionViewItem,
+    QWidget,
+    QVBoxLayout,
+    QProgressBar,
+)
+from qtpy.QtGui import QPainter, QColor
+from ert_shared.status.entity.state import REAL_STATE_TO_COLOR
+from ert_gui.model.snapshot import ProgressRole
+
+
+class ProgressView(QWidget):
+    def __init__(self, parent=None) -> None:
+        super(ProgressView, self).__init__(parent)
+
+        self._progress_tree_view = QTreeView(self)
+        self._progress_tree_view.setHeaderHidden(True)
+        self._progress_tree_view.setItemsExpandable(False)
+        self._progress_tree_view.setItemDelegate(ProgressDelegate(self))
+        self._progress_tree_view.setRootIsDecorated(False)
+        self._progress_tree_view.setFixedHeight(30)
+        self._progress_tree_view.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+
+        self._progress_bar = QProgressBar(self)
+        self._progress_bar.setRange(0, 0)
+        self._progress_bar.setFixedHeight(30)
+        self._progress_bar.setVisible(False)
+
+        layout = QVBoxLayout(self)
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.addWidget(self._progress_tree_view)
+        layout.addWidget(self._progress_bar)
+
+        self.setLayout(layout)
+
+        self.setFixedHeight(30)
+
+    def setModel(self, model) -> None:
+        self._progress_tree_view.setModel(model)
+
+    @Slot(bool)
+    def setIndeterminate(self, b: bool) -> None:
+        if b:
+            self._progress_bar.setVisible(True)
+            self._progress_tree_view.setVisible(False)
+        else:
+            self._progress_bar.setVisible(False)
+            self._progress_tree_view.setVisible(True)
+
+
+class ProgressDelegate(QStyledItemDelegate):
+    def __init__(self, parent=None) -> None:
+        super(ProgressDelegate, self).__init__(parent)
+
+        self.background_color = QColor(200, 210, 210)
+
+    def paint(self, painter, option: QStyleOptionViewItem, index: QModelIndex) -> None:
+        data = index.data(ProgressRole)
+        if data is None:
+            return
+
+        nr_reals = data["nr_reals"]
+        status = data["status"]
+        d = option.rect.width() / nr_reals
```

Comment:
This triggers `DeprecationWarning: an integer is required (got type float).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.` in Python 3.8 (the trigger happens on line 89, when it is actually passed to the painter). We should make the int conversion explicit.